### PR TITLE
Moving Instance Gateway to SIG-Network and renaming it

### PR DESCRIPTION
**Pull request by robscott**
New name will be gateway-api-inference-extension.

**Which issue(s) this PR fixes**:
None in this repo, but addresses https://github.com/kubernetes-sigs/llm-instance-gateway/issues/41.

For more context on the name change, refer to the original [proposal](https://docs.google.com/document/d/1gLYe9zPNeMuxKWolcu5MRm_Sq2xoZBXn100e-fZ6fX4/edit?tab=t.0) which has previously been discussed in SIG-Network, Instance Gateway, and Gateway API community meetings, as well as WG-Serving Slack and mailing list. 
  
Looking for LGTM from representatives from each of the following groups:

SIG Apps Leads:
/cc @janetkuo @kow3ns @soltysh 

SIG Net Leads:
/cc @aojea @danwinship @MikeZappa87 @shaneutt @thockin 

WG Serving Leads:
/cc @ArangoGutierrez @Jeffwan @SergeyKanzhelev @terrytangyuan

/hold for consensus

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
from sig-apps pov

File: sig-network/README.md
Comment (terrytangyuan) [inline sig-network/README.md:111]: Would you like to submit a request to rename the repo so that you can change the name in this PR?

**Comment by aojea** [comment]
/lgtm

This was presented during last sig network meeting with positive feedback

**Review by janetkuo — APPROVED** [review]
/lgtm

File: sig-network/README.md
Comment (BenTheElder) [reply sig-network/README.md:111]: It will be slightly easier to do in two steps, because then it's just SIG network request in kubernetes/org to rename the repo, and a follow-up PR here to update the repo name/link that we can trivially approve and merge without broad review.

Comment (BenTheElder) [reply sig-network/README.md:111]: Renaming the repo is a straightforward request as long as the name isn't uh ridiculous or something, but it's less straightforward if it's also tied up in switching the sponsoring group. I suggested we clarify the sponsoring SIG first and then do the rename because the SIG switch needs more people nominally involved than the rest.

Comment (robscott) [reply sig-network/README.md:111]: Good call, filed https://github.com/kubernetes/org/issues/5302 to track that

Comment (robscott) [reply sig-network/README.md:111]: Missed @BenTheElder's comments until after I'd made the k/org issue. I think it's useful to have this single reference point to capture approvals from leads. I'll plan on waiting for the rename to happen before merging this PR.

Comment (BenTheElder) [reply sig-network/README.md:111]: : shrug : we will merge it? :-)

up to you, either way works.

Comment (robscott) [reply sig-network/README.md:111]: To clarify, I left a hold on this PR, I meant that I'd remove that hold when the rename happens. The PR already has LGTM + approve labels so the only blocker is the hold.

**Review by Jeffwan — APPROVED** [review]
/lgtm

**Comment by ahg-g** [comment]
Anything still blocking this PR?

**Comment by BenTheElder** [comment]
IMHO: this could be unheld and a follow-up just to fix the name merged when the rename happens (pending https://github.com/kubernetes/org/issues/5302 ...)

**Comment by MadhavJivrajani** [comment]
I'm working on renaming the repo. Ideally, the rename in sigs.yaml would happen as part of that, but since this is already doing it, plase hold off on unholding the PR for now and I can unhold it once the rename is done!

File: sigs.yaml
Comment (BenTheElder) [inline sigs.yaml:2326]: TODO: fixup this link now that the repo was renamed?

(this link works, but we should use the canonical name)

Comment (robscott) [reply sigs.yaml:2326]: https://github.com/kubernetes/community/pull/8218

**Pull request merged**
